src/main/java/bio/terra/service/filedata/azure/tables/TableDependencyDao.java

```diff
@@ -75,19 +89,29 @@ private void createDependencyEntity(TableClient tableClient, UUID snapshotId, St
   public void deleteSnapshotFileDependencies(
       TableServiceClient tableServiceClient, UUID datasetId, UUID snapshotId) {
     String dependencyTableName = StorageTableName.DEPENDENCIES.toTableName(datasetId);
-    ;
+
     if (TableServiceClientUtils.tableHasEntries(tableServiceClient, dependencyTableName)) {
       TableClient tableClient = tableServiceClient.getTableClient(dependencyTableName);
       ListEntitiesOptions options =
           new ListEntitiesOptions().setFilter(String.format("snapshotId eq '%s'", snapshotId));
-      PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
-      var batchEntities =
-          entities.stream()
-              .map(entity -> new TableTransactionAction(TableTransactionActionType.DELETE, entity))
-              .collect(Collectors.toList());
-      logger.info(
-          "Deleting snapshot {} file dependencies from {}", snapshotId, dependencyTableName);
-      tableClient.submitTransaction(batchEntities);
+      List<TableEntity> entities =
+          tableClient.listEntities(options, null, null).stream().collect(Collectors.toList());
```

Comment:
So, I know it's generally not a good idea to load a ton of items into one list. I don't know that we're really looking at large enough numbers that would cause an issue here. 

If it's something worth pursuing, then maybe we could harness paged iterables to process the results of listEntites page by page. 
Maybe something like this? 
```
PagedIterable<TableEntity> entities = tableClient.listEntities(options, null, null);
 entities.streamByPage().forEach(resp -> {
     // then process the batch of entities? 
 });
```